sdk/src/main/java/com/bugsnag/android/BlockedThreadDetector.java

```diff
@@ -0,0 +1,106 @@
+package com.bugsnag.android;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.os.SystemClock;
+
+/**
+ * Detects whether a given thread is blocked by continuously posting a {@link Runnable} to it
+ * from a watcher thread, invoking a delegate if the message is not processed within
+ * a configured interval.
+ */
+final class BlockedThreadDetector {
+
+    private static final int DEFAULT_CHECK_INTERVAL_MS = 1000;
```

Comment:
As far as I can tell this interval is always used (apart from for testing).

This is slightly confusing because the name implies that it is only the default (and it perhaps could be configured by the user).

It also means that any threshold less than `DEFAULT_CHECK_INTERVAL_MS` will cause an `IllegalArgumentException` (which is not obvious)